kpagacz/wiki-on-fire
# feat: added backend documentation

* created a CloudFormation yaml with specifications for an S3 Bucket to host the docs
* created a GitHub workflow triggered on pushes to main to build and deployt docs to the S3 Bucket
* added a lot of documentation
* set up a prettier configuration file
* set up a JSDocs configuration file
Closes #168

Some quality of life fixes: running prettier should be now smooth - especially if you use the VSCode extension, prettier config includes a jsdoc plugin that adds support for formatting jsdocs - quite nice as well. You can ofc run prettier via the cli as well.

How to test:
* delete the test s3 bucket (if it's still there) that's on the aws right now (delete the files first, then delete the bucket)
* delete the test stack if there's any
* set up the bucket via the cloudformation yaml
* navigate to giddy-banana
* run `npm run docs`: it should build and open the documentation in a default browser
* test the workflow on github via manual dispatch of an action - it should upload the build documentation to the very same bucket created in the cloudformation yaml

This one is quite easy to test :)
